SimTKcommon/include/SimTKcommon/internal/Value.h

```diff
@@ -47,7 +51,33 @@ class AbstractValue {
     virtual String        getValueAsString() const = 0;
     virtual bool        isCompatible(const AbstractValue&) const = 0;
     virtual void compatibleAssign(const AbstractValue&) = 0;
-        
+
+    /// Retrieve the stored object as const ref -- read-only.
+    template<typename T>
+    const typename std::remove_reference<T>::type& getValue() {
+#ifndef NDEBUG
+      return dynamic_cast<const 
+			  Value<typename 
+				std::remove_reference<T>::type>&>(*this).thing;
+#else
+      return static_cast<const 
+			 Value<typename 
+			       std::remove_reference<T>::type>&>(*this).thing;
+#endif
+    }
+
+    /// Retrieve the stored object as ref -- read-write.
+    template<typename T>
+    typename std::remove_reference<T>::type& updValue() {
+#ifndef NDEBUG
+      return dynamic_cast<Value<typename 
+				std::remove_reference<T>::type>&>(*this).thing;
+#else
+      return static_cast<Value<typename 
+			       std::remove_reference<T>::type>&>(*this).thing;
+#endif
+    }
+
```

Comment:
The `remove_reference` utility is used here to account for users calling the function with reference types like `getValue<Foo&>`. In such cases, the reference is stripped.